lib/web/assistant.go

```diff
@@ -520,13 +525,29 @@ func (h *Handler) assistSSHExplainOutputLoop(ctx context.Context, assistClient *
 		return trace.Wrap(err)
 	}
 
-	//TODO(jakule): add token usage reporting when events are added to posthog
+	prompt, completion := model.CountTokens(tokenCount)
+
+	usageEventReq := &clientproto.SubmitUsageEventRequest{
+		Event: &usageeventsv1.UsageEventOneOf{
+			Event: &usageeventsv1.UsageEventOneOf_AssistAction{
+				AssistAction: &usageeventsv1.AssistAction{
+					Action:           actionSSHExplainCommand,
+					TotalTokens:      int64(completion + prompt),
+					PromptTokens:     int64(prompt),
+					CompletionTokens: int64(completion),
+				},
+			},
+		},
+	}
+	if err := usageRep.SubmitUsageEvent(ctx, usageEventReq); err != nil {
+		h.log.WithError(err).Warn("Failed to emit usage event")
+	}
```

Comment:
Should we report events synchronously? If somehting bad happens and this hangs or is slow, this will block execution. Also, if the context gets cancelled (e.g. the client disconnected) we won't report the event.

This discussion is ongoing here: https://github.com/gravitational/teleport/pull/30554#issuecomment-1682319427

Until a consensus is reached, I think we should align all assist reporting calls. Currently most of them are asynchornous with their own background context, but those new calls are synchronous.

Reply:
I made them synchronous for a different reason. I don't think we should wait for the request to be sent, as you mentioned, but this call requires auth client. Auth client can be closed as soon as the connection closes, so we may lose many events. 
I agree that we should agree on something, but I also don't think this is a blocker for the current implementation.